eks nodes fails with karpenter nodes

I haven't had time to troubleshoot this yet but I have a cluster with karpenter nodes
```
k get no                                                                                                                     
NAME                                             STATUS     ROLES    AGE     VERSION                                           
i-001b401a8f4cb3b48.us-east-2.compute.internal   Ready      <none>   61m     v1.27.3                                           
i-002c826ec1bc8c7c9.us-east-2.compute.internal   Ready      <none>   53m     v1.27.3                                           
i-003ca3cd1e27c400c.us-east-2.compute.internal   Ready      <none>   63m     v1.27.3                                           
i-003f808cae84dd6cb.us-east-2.compute.internal   Ready      <none>   69m     v1.27.3                                           
i-004125fe7b0156226.us-east-2.compute.internal   Ready      <none>   45m     v1.27.3
```
Trying to use the eks plugin gives me this error
```
k eks nodes                                                                                                                  
NAME     INSTANCE-TYPE     OS     CAPACITY-TYPE     ZONE     AMI-ID     AMI-NAME     AGE                                       
panic: runtime error: invalid memory address or nil pointer dereference                                                        
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3c739a6]                                                        
                                                                                                                               
goroutine 1 [running]:                                                                                                         
github.com/surajincloud/kubectl-eks/cmd.nodes(0x6193700?, {0x441a0f2?, 0x0?, 0x0?})                                            
        /home/runner/work/kubectl-eks/kubectl-eks/cmd/nodes.go:56 +0x886                                                       
github.com/spf13/cobra.(*Command).execute(0x6193700, {0x62d3ba8, 0x0, 0x0})                                                    
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862                                            
github.com/spf13/cobra.(*Command).ExecuteC(0x61939e0)                                                                          
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd                                           
github.com/spf13/cobra.(*Command).Execute(...)                                                                                 
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968                                                   
github.com/surajincloud/kubectl-eks/cmd.Execute()                                                                              
        /home/runner/work/kubectl-eks/kubectl-eks/cmd/root.go:25 +0x25                                                         
main.main()                                                                                                                    
        /home/runner/work/kubectl-eks/kubectl-eks/main.go:6 +0x17
```
I have 2 nodes in a node group
```
k eks nodegroups                                                                                                             
NAME     RELEASE             AMI_TYPE       INSTANCE_TYPES     STATUS                                                          
main     1.27.1-20230703     AL2_x86_64     t3.large           ACTIVE
```
version
```
k eks version
v0.4.1
```

--- a/cmd/nodes.go
+++ b/cmd/nodes.go
@@ -52,8 +52,14 @@ func nodes(cmd *cobra.Command, args []string) error {
 		// AMI
 		if i.Labels[kube.NodeGroupImage] == "" {
 			amiID = i.Labels[kube.KarpenterImage]
-			dis, _ := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{amiID}})
-			amiName = aws.ToString(dis.Images[0].Name)
+			if amiID != "" {
+				dis, err := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{amiID}})
+				if err != nil {
+					log.Fatal(err)
+				}
+				amiName = aws.ToString(dis.Images[0].Name)
+			}
+
 		} else {
 			amiID = i.Labels[kube.NodeGroupImage]
 			dis, err := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{amiID}})

--- a/cmd/nodes.go
+++ b/cmd/nodes.go
@@ -69,6 +69,22 @@ func nodes(cmd *cobra.Command, args []string) error {
 			amiName = aws.ToString(dis.Images[0].Name)
 		}
 
+		if amiID == "" {
+			instanceId := strings.Split(i.Spec.ProviderID, "/")[len(strings.Split(i.Spec.ProviderID, "/"))-1]
+			out, err := ec2Client.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
+				InstanceIds: []string{instanceId},
+			})
+			if err != nil {
+				log.Fatal(err)
+			}
+			amiID = *out.Reservations[0].Instances[0].ImageId
+			dis, err := ec2Client.DescribeImages(ctx, &ec2.DescribeImagesInput{ImageIds: []string{amiID}})
+			if err != nil {
+				log.Fatal(err)
+			}
+			amiName = aws.ToString(dis.Images[0].Name)
+		}
+
 		// Capacity Type
 		if i.Labels[kube.CapacityTypeLabel] != "" {
 			capacityType = i.Labels[kube.CapacityTypeLabel]

